src/lib/sql/sql_identifier_resolver.cpp

```diff
@@ -7,38 +7,43 @@ using namespace std::string_literals;  // NOLINT
 
 namespace opossum {
 
-void SQLIdentifierResolver::set_column_name(const std::shared_ptr<AbstractExpression>& expression,
+void SQLIdentifierResolver::add_column_name(const std::shared_ptr<AbstractExpression>& expression,
                                             const std::string& column_name) {
   auto& entry = _find_or_create_expression_entry(expression);
-  if (entry.identifier) {
-    entry.identifier->column_name = column_name;
-  } else {
-    entry.identifier.emplace(column_name);
-  }
+  entry.identifiers.emplace_back(column_name);
+}
+
+void SQLIdentifierResolver::reset_column_names(const std::shared_ptr<opossum::AbstractExpression>& expression) {
+  auto& entry = _find_or_create_expression_entry(expression);
```

Comment:
so, an Entry is potentially created if it didn't exist before. The function name doesn't reflect this. Do we even want this? If not, an Assert should be added, ensuring that an Entry for `expression` existed before. 
If we want it, the function name should reflect this behaviour and queries triggering this path should be added to the test suite.